ntbs-service/Services/UserSearchService.cs

```diff
@@ -2,48 +2,60 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Castle.Core.Internal;
+using MoreLinq;
 using ntbs_service.DataAccess;
 using ntbs_service.Models;
 using ntbs_service.Models.Entities;
 
 namespace ntbs_service.Services
 {
-    public interface ICaseManagerSearchService
+    public interface IUserSearchService
     {
-        Task<(IList<User> caseManagers, int count)> OrderAndPaginateQueryableAsync(
+        Task<(IList<User> users, int count)> OrderAndPaginateQueryableAsync(
             string searchKeyword,
             PaginationParametersBase paginationParameters);
     }
 
-    public class CaseManagerSearchService : ICaseManagerSearchService
+    public class UserSearchService : IUserSearchService
     {
         private readonly IReferenceDataRepository _referenceDataRepository;
+        private readonly IUserRepository _userRepository;
 
-        public CaseManagerSearchService(IReferenceDataRepository referenceDataRepository)
+        public UserSearchService(IReferenceDataRepository referenceDataRepository, IUserRepository userRepository)
         {
             _referenceDataRepository = referenceDataRepository;
+            _userRepository = userRepository;
         }
 
-        public async Task<(IList<User> caseManagers, int count)> OrderAndPaginateQueryableAsync(
+        public async Task<(IList<User> users, int count)> OrderAndPaginateQueryableAsync(
             string searchKeyword,
             PaginationParametersBase paginationParameters)
         {
             var searchKeywords = searchKeyword.Split(" ")
                 .Where(x => !x.IsNullOrEmpty())
                 .Select(s => s.ToLower()).ToList();
-            var caseManagers = await _referenceDataRepository.GetAllCaseManagersOrdered();
+            
+            var allPhecs = await _referenceDataRepository.GetAllPhecs();
+            var filteredPhecs = allPhecs.Where(phec => searchKeywords.Any(s => phec.Name.ToLower().Contains(s)));
+
+            var caseManagersAndRegionalUsers = (await _referenceDataRepository.GetAllCaseManagersOrdered())
+                .Concat(_userRepository.GetUserQueryable().ToList()
+                    .Where(user => user.AdGroups != null && allPhecs.Any(phec => user.AdGroups.Split(",").Contains(phec.AdGroup))))
```

Comment:
Do we need this `Where` clause, as it's repeated below?

Reply:
This where clause is taking all users with ANY regional membership. This was to limit the amount of users I am concatenating with the case managers, because we may want a regional user (that isn't a case manager) that has a matching name (given, family or display).

Reply:
Fair enough. After you've removed the reference data repo call, I think the affect of this will be pretty minimal, so we should probably drop it. (Unless I'm missing something and you think it will have a decent performance impact?)